packages/agent-utils-fs/src/FileSystemWorkspace.ts

```diff
@@ -71,4 +74,37 @@ export class FileSystemWorkspace {
     const absPath = this.toWorkspacePath(subpath);
     fs.appendFileSync(absPath, data);
   }
+
+  async exec(command: string, args?: string[]): Promise<{ exitCode: number; stdout: string; stderr: string }> {
+    return await new Promise((resolve, reject) => {
+      const toExec = args ? `${command} ${args.join(" ")}` : command;
+      const child = spawn(toExec, { cwd: this._workspacePath });
+
+      let stdout = "";
+      let stderr = "";
+
+      child.on("error", (error: Error) => {
+        reject(error);
+      });
+
+      child.stdout?.on("data", (data: string) => {
+        stdout += data.toString();
+      });
+
+      child.stderr?.on("data", (data: string) => {
+        stderr += data.toString();
+      });
+
+      child.on("exit", (exitCode: number) => {
+        resolve({ exitCode, stdout, stderr });
+      });
+    });
+  }
+
+  async poetryInit(): Promise<void> {
+    await this.exec("poetry", ["init", "-n"]);
+    const dependencies = getPythonDependencies(this._workspacePath);
```

Comment:
Very strange to have this here, doesn't match the responsibilities of the "workspace"

Reply:
Thanks for the feedback. Should we rename the `Workspace` interface to something more specific? I didn't think it seemed "very strange" to regard the concept of initializing and managing a programming environment to be within the responsibilities of something called "Workspace", but I agree it seems wrong if you regard the Workspace to be something more narrow or low level.

Reply:
Yah the workspace is just the user's workspace (working directory). All other use-case specific things are outside of this, and simply use this abstraction. All use-case specific things can be put within agent functions.

Reply:
Awesome. Well, I've moved the poetry logic to an agent function. I believe this particular concern is resolved.